component/src/main/java/io/siddhi/extension/script/js/EvalFunctionExtension.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.siddhi.extension.script.js;
+
+import io.siddhi.annotation.Example;
+import io.siddhi.annotation.Extension;
+import io.siddhi.annotation.Parameter;
+import io.siddhi.annotation.ParameterOverload;
+import io.siddhi.annotation.ReturnAttribute;
+import io.siddhi.annotation.util.DataType;
+import io.siddhi.core.config.SiddhiQueryContext;
+import io.siddhi.core.exception.SiddhiAppRuntimeException;
+import io.siddhi.core.executor.ConstantExpressionExecutor;
+import io.siddhi.core.executor.ExpressionExecutor;
+import io.siddhi.core.executor.function.FunctionExecutor;
+import io.siddhi.core.util.config.ConfigReader;
+import io.siddhi.core.util.snapshot.state.State;
+import io.siddhi.core.util.snapshot.state.StateFactory;
+import io.siddhi.query.api.definition.Attribute;
+import io.siddhi.query.api.exception.SiddhiAppValidationException;
+import jdk.nashorn.internal.runtime.ParserException;
+
+import java.util.Locale;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import static io.siddhi.query.api.definition.Attribute.Type.STRING;
+
+/**
+ * eval(string, dataType)
+ * Evaluate the string and return output as dataType
+ * Accept Type(s): (STRING, STRING)
+ * Return Type(s): STRING/INT/LONG/FLOAT/DOUBLE/BOOL
+ */
+@Extension(
+        name = "eval",
+        namespace = "script",
+        description = "This extension evaluated a given string and " +
+                "return the output according to the user specified data type.",
+        parameters = {
+                @Parameter(name = "expression",
+                        description = "The string with an logical or arithmetic expression.",
+                        type = {DataType.STRING},
+                        dynamic = true),
+                @Parameter(name = "data.type",
+                        description = "The return type of the evaluated expression." +
+                                " Supported types are int/long/float/double/bool/string.",
+                        type = {DataType.STRING},
+                        dynamic = true),
```

Comment:
This should be static as if the return type varies for each event then there can be compilation issues, in your case you are setting this for OBJECT which is also wrong